In [7]:
import m_and_m
import pandas as pd

In [2]:
mix94 = dict(brown=30,
             yellow=20,
             red=20,
             green=10,
             orange=10,
             tan=10)


mix96 = dict(blue=24,
             green=20,
             orange=16,
             yellow=14,
             red=13,
             brown=13)

In [3]:
hypoA = dict(bag1=mix94, bag2=mix96)
hypoB = dict(bag1=mix96, bag2=mix94)

hypoA represents the hypothesis that Bag 1 is from 1994 and Bag 2 from 1996. hypoB is the other way around..

Hence, for simplecity ..

In [4]:
hypotheses = dict(A=hypoA, B=hypoB)

## Likelihood function:

In [6]:
def Likelihood(data, hypo):
    bag, color = data
    mix = self.hypotheses[hypo][bag]
    like = mix[color]
    return like

## Calling m_and_m:

In [8]:
m_and_m.main()

A 0.740740740741
B 0.259259259259


The posterior probability of A is about 0.74 and for B is about 0.26 !